account/account.go

```diff
@@ -824,3 +824,55 @@ func FmtCalldataCairo2(fnCalls []rpc.FunctionCall) []*felt.Felt {
 
 	return execCallData
 }
+
+type DeployOptions struct {
+	ClassHash           *felt.Felt    // ClassHash is the unique identifier of the account class in the network.
+	MaxFee              *felt.Felt    // MaxFee represents the maximum fee the user is willing to pay for the deployment transaction.
+	DeployWaitTime      time.Duration // DeployWaitTime specifies the duration to wait for the deployment transaction to be processed.
+	ConstructorCalldata []*felt.Felt  // ConstructorCalldata contains the calldata to be passed to the constructor of the account contract upon deployment.
+}
+
+func (account *Account) CreateAndExecuteAddDeployAccount(options DeployOptions) (*rpc.AddDeployAccountTransactionResponse, error) {
```

Comment:
This works for rpcv05 and lower. However, rpcv06 introduces a new deployAccount transaction version (v3), meaning users won't be able to use this method to submit v3 transactions. It would be great if we could make this compatible with both transaction versions (eg by accepting the transaction as an argument, tx rpc.DeployAccountType)

Reply:
sure will fix it

Reply:
compatible with both rpcv06 and previous, or both as in v1 and v2 version?

Reply:
by "this method" you mean the whole method "CreateAndExecuteAddDeployAccount" or the way you deploy accounts like in the function?

Reply:
The CreateAndExecuteAddDeployAccount method.